language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/documentsymbol/DocumentSymbolResolver.java

```diff
@@ -145,24 +179,185 @@
+        SymbolKind symbolKind = SymbolKind.Object;
         Range range = DocumentSymbolUtil.generateNodeRange(serviceDeclarationNode);
         Optional<MetadataNode> metadata = serviceDeclarationNode.metadata();
         boolean isDeprecated = metadata.isPresent() &&
                 DocumentSymbolUtil.isDeprecated(metadata.get());
         List<DocumentSymbol> children = transformMembers(serviceDeclarationNode.members());
-        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
-                range, range, isDeprecated, children, this.context));
+        return Optional.of(createDocumentSymbol(name.toString(), symbolKind, range, range, isDeprecated, children));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(TypeDefinitionNode typeDefinitionNode) {
+        String name = typeDefinitionNode.typeName().text();
+        Node typeDescriptor = typeDefinitionNode.typeDescriptor();
+        SymbolKind symbolKind;
+        List<DocumentSymbol> children = new ArrayList<>();
+        switch (typeDescriptor.kind()) {
+            case RECORD_TYPE_DESC:
+                symbolKind = SymbolKind.Struct;
+                RecordTypeDescriptorNode recordTypeDescriptorNode = (RecordTypeDescriptorNode) typeDescriptor;
+                children.addAll(transformMembers(recordTypeDescriptorNode.fields()));
+                Optional<RecordRestDescriptorNode> restTypeDec = recordTypeDescriptorNode.recordRestDescriptor();
+                if (restTypeDec.isPresent()) {
+                    Optional<DocumentSymbol> restDocSymbol = restTypeDec.get().apply(this);
+                    restDocSymbol.ifPresent(children::add);
+                }
+                break;
+            case OBJECT_TYPE_DESC:
+                symbolKind = SymbolKind.Interface;
+                children.addAll(transformMembers(((ObjectTypeDescriptorNode) typeDescriptor).members()));
+                break;
+            default:
+                symbolKind = SymbolKind.TypeParameter;
+        }
+        Range range = DocumentSymbolUtil.generateNodeRange(typeDefinitionNode);
+        Optional<MetadataNode> metadata = typeDefinitionNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.of(createDocumentSymbol(name, symbolKind, range, range, isDeprecated, children));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ModuleVariableDeclarationNode moduleVariableDeclarationNode) {
+        BindingPatternNode bindingPatternNode = moduleVariableDeclarationNode.typedBindingPattern().bindingPattern();
+        //Only capture binding pattern is considered. Wildcard and other binding patterns are ignored.
+        String name = bindingPatternNode.kind() == SyntaxKind.CAPTURE_BINDING_PATTERN ?
+                bindingPatternNode.toSourceCode() : "";
+        if (name.isEmpty()) {
+            return Optional.empty();
+        }
```

Comment:
Use the kind check here. No need to proceed with the string. We can use early return.

Reply:
Done